### PR TITLE
Refactor tools to improve Bugzilla data

### PR DESCRIPTION
**Pull request by calixteman**

### CODE REVIEW

File: auto_nag/bzcleaner.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+    def amend_bzparams(self, params, bug_ids):
+        """Amend the Bugzilla params"""
+        if 'include_fields' in params:
+            fields = params['include_fields']
+            if isinstance(fields, list):
+                if 'id' not in fields:
+                    fields.append('id')
+            elif isinstance(fields, six.string_types):
+                if fields != 'id':
```
Comment (sylvestre) [inline auto_nag/bzcleaner.py]: Seems like duplicate code?

File: auto_nag/bzcleaner.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        else:
+            params['include_fields'] = 'id'
+
+        if bug_ids:
+            params['bug_id'] = bug_ids
+
+    def get_bugs(self, date='today', bug_ids=[]):
+        """Get the bugs"""
+        # the search query can be long to evaluate
+        TIMEOUT = 240
```
Comment (sylvestre) [inline auto_nag/bzcleaner.py]: Maybe we should move that in the configuration file?

File: auto_nag/bzcleaner.py (hunk 1 above)
Comment (calixteman) [reply auto_nag/bzcleaner.py]: why ?

**Comment by calixteman** [comment]
Don't merge I've few modifs to do about config stuff

File: auto_nag/bzcleaner.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    def send_email(self, date='today', dryrun=False):
+        """Send the email"""
+        login_info = self.get_login_info()
+        if date:
+            date = lmdutils.get_date(date)
+        title, body = self.get_email(login_info['bz_api_key'], date, dryrun)
+        if title:
+            mail.send(login_info['ldap_username'],
+                      utils.get_config('common', 'receivers', [
+                                       'sylvestre@mozilla.com']),
```
Comment (sylvestre) [inline auto_nag/bzcleaner.py]: Maybe move that into the conf?

**Comment by sylvestre** [comment]
Bravo, this is much better!

**Comment by sylvestre** [comment]
> Coverage increased (+5.9%) to 47.364% when pulling e496370 on calixteman:refactor_bz_tools into 9328048 on mozilla:master.

Bravo!

**Pull request merged**